src/api/eth_filter.rs

```diff
@@ -1,108 +1,138 @@
 //! `Eth` namespace, filters.
 
+
 use std::marker::PhantomData;
-use std::ops::Deref;
+use std::time::Duration;
+use serde::de::DeserializeOwned;
+use tokio_timer::Timer;
+use futures::{Poll, Future, Stream, stream};
 
 use api::Namespace;
-use futures::{Async, Poll, Future};
 use helpers::{self, CallResult};
 use types::{Filter, H256, Log, U256};
-use {Transport, Error};
+use {Transport, Error, rpc};
+
+/// Waits for X consecutive confirmations that when `validate` has happened
+fn wait_for_confirmations_internal<'a, T, F, V>(transport: T, confirmations: usize, validate: &'a V)
+  -> Box<Future<Item = (), Error = Error> + 'a> where
+  T: 'a + Transport + Clone,
+  F: 'a + Future<Item = bool, Error = Error>,
+  V: 'a + Fn(&H256) -> F,
+{
+  let eth = EthFilter::new(transport);
+  let result = eth.create_blocks_filter()
+    .and_then(move |filter| {
+      filter.stream(Duration::from_secs(1))
+        .skip_while(move |hash| validate(hash).map(|ok| !ok))
+        .skip(1)
+        .take_while(validate)
+        .take(confirmations as u64)
+        .collect()
+        .and_then(move |hashes| if hashes.len() == confirmations {
```

Comment:
Currently, parity does not inform about forks happening. The next hash is always a hash of the `previous_block_number + 1`, even if there was a fork in the meantime. If event happened on both chains, let's not bother with possible difference in confirmation counter.

The different approach to the problem, would be polling `eth_blockNumber`, but it also has it's cons